src/Tribe/Views/V2/Query/Query.php

```diff
@@ -0,0 +1,225 @@
+		  -- End after the period start.
+		  AND end_date.meta_value >= %s
+		  AND (all_day.post_id IS NULL OR all_day.meta_value = 'yes'); ";
+
+		$results = $wpdb->get_results(
+			$wpdb->prepare(
+				$query,
+				$use_site_timezone ? '_EventStartDateUTC' : '_EventStartDate',
+				$end->format( Dates::DBDATETIMEFORMAT ),
+				$use_site_timezone ? '_EventSTartDateUTC' : '_EventEndDate',
+				$start->format( Dates::DBDATETIMEFORMAT )
+			),
+			ARRAY_A
+		);
+
+		try {
+			$one_day = new \DateInterval( 'P1D' );
+		} catch ( \Exception $e ) {
+			// This should not happen, but let's make sure.
+			return;
+		}
+
+		$request_period = new \DatePeriod( $start, $one_day, $end );
+
+		$grouped_by_start_date = [];
+		if ( ! empty( $results ) ) {
+			$site_timezone = Timezones::build_timezone_object();
+
+			// @todo refactor this into smaller methods.
+			$grouped_by_start_date = array_reduce( $results,
+				static function ( array $buffer, array $result ) use ( $use_site_timezone, $site_timezone, $one_day )
+				{
+					$display_timezone = $use_site_timezone
+						? $site_timezone
+						: Timezones::build_timezone_object( $result['timezone'] );
+					$start_date       = Dates::build_date_object( $result['start_date'], $display_timezone );
+					$end_date         = Dates::build_date_object( $result['end_date'], $display_timezone );
+					if (
+						$start_date->format( Dates::DBDATEFORMAT ) === $end_date->format( Dates::DBDATEFORMAT )
+					) {
+						$overlapping_days = [ $start_date->format( Dates::DBDATEFORMAT ) ];
+					} else {
+						/*
+						 * "Move" the end date, adding a day to it, to make sure the end date is included in the period.
+						 * Else multi-day events would only overlap the first two dates.
+						 */
+						$moved_end_date = clone $end_date;
+						$moved_end_date->add( $one_day );
+						$period           = new \DatePeriod( $start_date, $one_day, $moved_end_date );
+						$overlapping_days = [];
+						/** @var \DateTimeInterface $d */
+						foreach ( $period as $d ) {
+							// This is skipping the end day on multi-day events.
+							$overlapping_days[] = $d->format( Dates::DBDATEFORMAT );
+							// Sanity check: break when the current day is equal to the event end date.
+							$reached_end = $d->format( Dates::DBDATEFORMAT ) === $end_date->format( Dates::DBDATEFORMAT );
+							if ( $reached_end ) {
+								break;
+							}
+						}
+					}
+
+					// Normalize the timezone to the site one.
+					$result['start_date'] = $start_date->setTimezone( $site_timezone )->format( 'Y-m-d H:i:s' );
+					$result['end_date']   = $end_date->setTimezone( $site_timezone )->format( 'Y-m-d H:i:s' );
+
+					foreach ( $overlapping_days as $overlap_day ) {
+						if ( isset( $buffer[ $overlap_day ] ) ) {
+							$buffer[ $overlap_day ][] = new Event_Result( $result );
+						} else {
+							$buffer[ $overlap_day ] = [ new Event_Result( $result ) ];
+						}
+					}
+
+					return $buffer;
+				}, [] );
+		}
+
```

Comment:
Filter before setting to cache.

Reply:
Once this goes into a `caching` repository implementation, the default repository extension and filtering system will apply.